fedimint-core/src/lib.rs

```diff
@@ -24,33 +42,56 @@ pub use crate::core::server;
 use crate::encoding::{Decodable, DecodeError, Encodable};
 use crate::module::registry::ModuleDecoderRegistry;
 
+/// Admin (guardian) client types
 pub mod admin_client;
+/// Client API request handling
 pub mod api;
+/// Federation-stored client backups
 pub mod backup;
+/// Gradual bitcoin dependency migration helpers
 pub mod bitcoin_migration;
 pub mod bitcoinrpc;
 pub mod cancellable;
+/// Federation configuration
 pub mod config;
+/// Fundamental types
 pub mod core;
+/// Database handling
 pub mod db;
+/// Consensus encoding
 pub mod encoding;
 pub mod endpoint_constants;
+/// Common environment variables
 pub mod envs;
 pub mod epoch;
+/// Formatting helpers
 pub mod fmt_utils;
+/// Hex encoding helpers
 pub mod hex;
+/// Common macros
 #[macro_use]
 pub mod macros;
+/// Extenable module sysystem
 pub mod module;
+/// Peer networking
 pub mod net;
+/// Client query system
 pub mod query;
+/// Task handling, including wasm safe logic
 pub mod task;
+/// Types handling per-denomination values
 pub mod tiered;
+/// Types handling multiple per-denomination values
 pub mod tiered_multi;
+/// Time handling, wasm safe functionality
 pub mod time;
+/// Timing helpers
 pub mod timing;
+/// Fedimint transaction (inpus + outputs + signature) types
 pub mod transaction;
+/// Peg-in txo proofs
 pub mod txoproof;
```

Comment:
not blocking, but maybe this should move out of `fedimint-core` at some point?

Reply:
Added. https://github.com/fedimint/fedimint/issues/4691